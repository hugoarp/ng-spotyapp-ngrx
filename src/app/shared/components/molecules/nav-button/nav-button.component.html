<a class="nav-button">
  <app-g-icon [icon]="icon"></app-g-icon>
  <app-avatar [avatar]="avatar" *ngIf="avatar"></app-avatar>
  <app-text-a [text]="label"></app-text-a>
</a>
