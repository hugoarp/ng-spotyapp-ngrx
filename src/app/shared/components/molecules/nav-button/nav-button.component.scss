@import "variables";
@import "mixins";

.nav-button {
  display: flex;
  flex-direction: column;
  align-items: center;
  color: $gray-primary;
  padding: 0.5rem;
  @include from($medium) {
    flex-direction: row;
    gap: 0.5rem;
  }
  &:hover {
    color: $light-primary;
    cursor: default;
  }
}
