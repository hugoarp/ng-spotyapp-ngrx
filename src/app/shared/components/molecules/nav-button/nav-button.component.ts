import { Component, Input, OnInit } from '@angular/core';
import { Button } from 'src/app/core/interfaces/button';

@Component({
  selector: 'app-nav-button',
  templateUrl: './nav-button.component.html',
  styleUrls: ['./nav-button.component.scss'],
})
export class NavButtonComponent implements OnInit {
  @Input() buttonInfo: Button = {
    icon: '',
    label: '',
    route: '',
  };

  constructor() {}

  ngOnInit(): void {}
}
