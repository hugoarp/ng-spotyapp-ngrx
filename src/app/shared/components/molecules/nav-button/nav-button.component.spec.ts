import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';

import { NavButtonComponent } from './nav-button.component';

describe('NavButtonComponent', () => {
  let component: NavButtonComponent;
  let fixture: ComponentFixture<NavButtonComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [NavButtonComponent],
      schemas: [CUSTOM_ELEMENTS_SCHEMA],
    }).compileComponents();

    fixture = TestBed.createComponent(NavButtonComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  test('debe hacer match con el snapshot', () => {
    expect(compiled).toMatchSnapshot();
  });
});
