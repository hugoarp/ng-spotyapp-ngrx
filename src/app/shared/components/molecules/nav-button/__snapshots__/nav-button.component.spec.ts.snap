// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavButtonComponent debe hacer match con el snapshot 1`] = `
<div
  id="root1"
>
  <a
    class="nav-button"
  >
    <app-g-icon />
    <app-text-a />
  </a>
</div>
`;
